src/Command/Branch/BranchChangelogCommand.php

```diff
@@ -45,44 +62,57 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        $branch = $input->getArgument('branch') ?: $this->getHelper('git')->getActiveBranchName();
+
         try {
-            $latestTag = $this->getHelper('git')->getLastTagOnBranch();
+            $latestTag = $this->getHelper('git')->getLastTagOnBranch($branch);
         } catch (\RuntimeException $e) {
-            $output->writeln('<info>There were no tags found</info>');
+            $this->getHelper('gush_style')->note(
+                sprintf('No tags were found on branch "%s".', $branch)
+            );
 
             return self::COMMAND_SUCCESS;
         }
 
         $adapter = $this->getIssueTracker();
-        $commits = $this->getHelper('git')->getLogBetweenCommits($latestTag, 'HEAD');
-
-        // Filter commits that reference an issue
-        $issues = [];
-
-        foreach ($commits as $commit) {
-            // Cut issue id from branch name (merge commits)
-            if (preg_match('/\/#([0-9]+)/i', $commit['message'], $matchesGush) && isset($matchesGush[1])) {
-                $issues[] = $matchesGush[1];
+        $commits = $this->getHelper('git')->getLogBetweenCommits($latestTag, $branch);
+        $issues = $this->getIssuesFromCommits($commits, $input->getOption('search'));
+
+        foreach ($issues as $id => $idLabel) {
+            // ignore missing issues
+            try {
+                $issue = $adapter->getIssue($id);
+            } catch (\Exception $e) {
+                continue;
             }
 
-            // Cut issue id from commit message
-            if (preg_match('/[close|closes|fix|fixes] #([0-9]+)/i', $commit['message'], $matchesGitRepo)
-                && isset($matchesGitRepo[1])
-            ) {
-                $issues[] = $matchesGitRepo[1];
-            }
+            $output->writeln(
+                sprintf("%s: %s   <info>%s</info>", $idLabel, $issue['title'], $issue['url'])
```

Comment:
we are using single quotes everywhere, why this needs double quotes? And Why the arbitrary space after the second string placeholder?